### PR TITLE
ENT-11175: Rewrote the intro to be more clear about different types of projects

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: JSON.md [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+See the [README](https://github.com/cfengine/cfbs/blob/master/README.md) for general documentation on the `cfbs` tool and its commands.
+This file is specifically about the `cfbs.json` file format, it should serve as detailed and useful information for more advanced users who are making their own modules or contributing to CFEngine Build.
 
-All these 3 share 1 standard format, commonly called `cfbs.json`.
+There are 3 types of `cfbs.json` files:
 
-(There is also [`versions.json`](https://github.com/cfengine/cfbs-index/blob/master/versions.json), but this specification is not about that file).
+* For building a policy set you create a project with type `policy-set`.
+  This is the default when running `cfbs init`, and what most users encounter when first using the tool and CFEngine.
+  You then need to use the `build` key to specify which modules to use in `cfbs build`.
```
Comment (craigcomstock) [inline JSON.md:14]:
```suggestion
  You then need to use the top-level `build` key to specify which modules to use in `cfbs build`.
```

File: JSON.md [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
-* The index of all available modules which is available on GitHub
-* A `cfbs.json` file with necessary metadata for adding a module using URL (not from index)
+See the [README](https://github.com/cfengine/cfbs/blob/master/README.md) for general documentation on the `cfbs` tool and its commands.
+This file is specifically about the `cfbs.json` file format, it should serve as detailed and useful information for more advanced users who are making their own modules or contributing to CFEngine Build.
 
-All these 3 share 1 standard format, commonly called `cfbs.json`.
+There are 3 types of `cfbs.json` files:
 
-(There is also [`versions.json`](https://github.com/cfengine/cfbs-index/blob/master/versions.json), but this specification is not about that file).
+* For building a policy set you create a project with type `policy-set`.
```
Comment (craigcomstock) [inline JSON.md:12]:
```suggestion
* For building a policy set you create a project with a top-level key `type` with the value `policy-set`.
```

File: JSON.md [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+  This is the default when running `cfbs init`, and what most users encounter when first using the tool and CFEngine.
+  You then need to use the `build` key to specify which modules to use in `cfbs build`.
+* The index of all available modules for `cfbs add <module-name>` has type `index`.
```
Comment (craigcomstock) [inline JSON.md:15]:
```suggestion
* For specifying all available modules for `cfbs add <module-name>` add a top-level key `type` with value `index`.
```

File: JSON.md [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
+  The available modules must be in a dictionary in the `index` field.
+  By default, [this index available GitHub](https://github.com/cfengine/build-index/blob/master/cfbs.json) is used.
+* When developing your own reusable modules to use in other projects, `module` type is used.
```
Comment (craigcomstock) [inline JSON.md:18]:
```suggestion
* When developing your own reusable modules to use in other projects, a top-level `type` with value `module` is used.
```

File: JSON.md [hunk 5]
```diff
@@ -1,14 +1,27 @@
...
+
+When `cfbs` is using the default index and when we build the [build.cfengine.com](https://build.cfengine.com) website, we use a separate [`versions.json`](https://github.com/cfengine/build-index/blob/master/versions.json) file to keep track of all the versions of modules, their tarballs and checksums.
+When contributors edit the index ([like this](https://github.com/cfengine/build-index/pull/465/files)), an automated PR is generated to make the appropriate edit to `versions.json` ([like this](https://github.com/cfengine/build-index/pull/466/files)), (after downloading and uploading the module), so users don't have to `versions.json` manually.
```
Comment (craigcomstock) [inline JSON.md:21]:
```suggestion
When contributors edit the index ([like this](https://github.com/cfengine/build-index/pull/465/files)), an automated PR is generated to make the appropriate edit to `versions.json` ([like this](https://github.com/cfengine/build-index/pull/466/files)), (after downloading and uploading the module), so users don't have to update `versions.json` manually.
```

File: JSON.md [hunk 6]
```diff
@@ -1,14 +1,27 @@
...
+The _policy set_ is what you deploy to your CFEngine hub / policy server, in `/var/cfengine/masterfiles`.
 
-* The project file (`cfbs.json`) generated by running `cfbs init` when you start a new project
-* The index of all available modules which is available on GitHub
-* A `cfbs.json` file with necessary metadata for adding a module using URL (not from index)
+See the [README](https://github.com/cfengine/cfbs/blob/master/README.md) for general documentation on the `cfbs` tool and its commands.
+This file is specifically about the `cfbs.json` file format, it should serve as detailed and useful information for more advanced users who are making their own modules or contributing to CFEngine Build.
 
-All these 3 share 1 standard format, commonly called `cfbs.json`.
+There are 3 types of `cfbs.json` files:
```
Comment (craigcomstock) [inline JSON.md:10]:
```suggestion
The type of a project is specified in a top-level `type` key in `cfbs.json`.
There are 3 types of projects:
```

**Review by craigcomstock — COMMENTED** [review]
some suggestions.

Reading it initially I didn't understand exactly what you were talking about until I saw some more context like the example. I think it helps to be more specific up-front about "top-level key `type`".

File: JSON.md (hunk 1 above)
Comment (olehermanse) [reply JSON.md:14]: Hmm, to me it seems to be a bit more tedious / harder to read if that is repeated every time. Probably I'll try to accept some, but not all, of the suggestions.

**Review by craigcomstock — APPROVED** [review]

File: JSON.md [hunk 7]
```diff
@@ -1,14 +1,28 @@
...
+The type of a project is specified in a top-level `type` key in `cfbs.json`.
+There are 3 types of projects:
 
-(There is also [`versions.json`](https://github.com/cfengine/cfbs-index/blob/master/versions.json), but this specification is not about that file).
+* For building a policy set you create a project with type `policy-set`.
+  This is the default when running `cfbs init`, and what most users encounter when first using the tool and CFEngine.
+  You then need to use the top-level `build` key to specify which modules to use in `cfbs build`.
+* The index of all available modules for `cfbs add <module-name>` has type `index`.
+  The available modules must be in a dictionary in the `index` field.
+  By default, [this index available GitHub](https://github.com/cfengine/build-index/blob/master/cfbs.json) is used.
```
Comment (larsewi) [inline JSON.md:18]: See https://github.com/cfengine/cfbs/pull/179/files

**Review by larsewi — COMMENTED** [review]
Great 🚀

File: JSON.md [hunk 8]
```diff
@@ -1,14 +1,28 @@
...
-* A `cfbs.json` file with necessary metadata for adding a module using URL (not from index)
+See the [README](https://github.com/cfengine/cfbs/blob/master/README.md) for general documentation on the `cfbs` tool and its commands.
+This file is specifically about the `cfbs.json` file format, it should serve as detailed and useful information for more advanced users who are making their own modules or contributing to CFEngine Build.
 
-All these 3 share 1 standard format, commonly called `cfbs.json`.
+The type of a project is specified in a top-level `type` key in `cfbs.json`.
+There are 3 types of projects:
 
-(There is also [`versions.json`](https://github.com/cfengine/cfbs-index/blob/master/versions.json), but this specification is not about that file).
+* For building a policy set you create a project with type `policy-set`.
```
Comment (oleorhagen) [inline JSON.md:13]: I would suggest enumarting these, starting with the key name. It means I don't have to read the whole sentence to look up which key we are talking about, and it is thus easier to use as a lookup table of sorts

Comment (olehermanse) [reply JSON.md:13]: To me that seems more appropriate for the keys section below. In the intro I prefer just sentences.

Comment (olehermanse) [reply JSON.md:13]: https://github.com/cfengine/cfbs/pull/180

**Pull request merged**